lib/hammer_cli_foreman/compute_resource.rb

```diff
@@ -70,24 +72,15 @@ def print_data(data)
 
     class CreateCommand < HammerCLIForeman::CreateCommand
 
-      VALIDATION_PER_PROVIDER = {
-        'libvirt'   => [:option_url],
-        'ovirt'     => [:option_url, :option_user, :option_password, :option_datacenter],
-        'openstack' => [:option_url, :option_user, :option_password],
-        'racksapce' => [:option_url],
-        'ec2'       => [:option_region, :option_user, :option_password],
-        'vmware'    => [:option_user, :option_password, :option_datacenter, :option_server]
-      }
-
       success_message _("Compute resource created.")
       failure_message _("Could not create the compute resource")
 
       build_options
 
       validate_options do
-        required_options = (VALIDATION_PER_PROVIDER[
+        required_options = (::HammerCLIForeman.compute_resources[
           option(:option_provider).value.to_s.downcase
-        ] || []).unshift(:option_name, :option_provider).uniq
+        ] || []).compute_resource_creation_validation
```

Comment:
We've lost `:option_name` and `:option_provider` was it intentional? If not can we add it to the base compute_resource and extend in successors?